domain/machine/state/state.go

```diff
@@ -411,6 +413,120 @@ WHERE st.machine_uuid = $machineUUID.uuid;
 	}, nil
 }
 
+// SetRunningAgentBinaryVersion sets the running agent binary version for the
+// provided machine uuid. Any previously set values for this machine uuid will
+// be overwritten by this call.
+//
+// The following errors can be expected:
+// - [machineerrors.MachineNotFound] if the machine does not exist.
+// - [coreerrors.NotSupported] if the architecture is not known to the database.
+func (st *State) SetRunningAgentBinaryVersion(
+	ctx context.Context,
+	machineUUID string,
+	version coreagentbinary.Version,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	type ArchitectureMap struct {
+		ID   int    `db:"id"`
+		Name string `db:"name"`
+	}
+	archMap := ArchitectureMap{Name: version.Arch}
+
+	archMapStmt, err := st.Prepare(`
+SELECT id AS &ArchitectureMap.id FROM architecture WHERE name = $ArchitectureMap.name
+`, archMap)
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	type MachineAgentVersion struct {
+		MachineUUID    string `db:"machine_uuid"`
+		Version        string `db:"version"`
+		ArchitectureID int    `db:"architecture_id"`
```

Comment:
Question: Why do we have a separate `architecture_id` here rather than using the one from `machine_cloud_instance`?  I assume we'd require the match at the very least.

I do plan to update the `arch` in `machine_cloud_instance` to reference the `architecture` table.

Reply:
Good question. Ian asked the same question. I think it would be wrong to do this at the moment. This is what is reported by the client upwards.

We should avoid correlating the two together as gospel because that linkage is done else where. We also have other entities that follow this pattern and can run over different arch's based on the scheduler. Think Kubernetes.

My two cents is avoid the temptation to link these together as one is asking "What should I be running" and the other is saying "This is what I am running".

Reply:
Please add a comment on this for our future selves.

Reply:
Done & Done